app/tray/Account/Requests/TransactionRequest/TxMainNew/overview.js

```diff
@@ -85,12 +85,12 @@ const TxOverview = ({
   simple,
   valueColor
 }) => {
-  const { recipientType, decodedData: { method } = {}, data: tx = {} } = req
+  const { decodedData: { method } = {}, data: tx = {} } = req
   const { to, value, data: calldata } = tx
 
   const isContractDeploy = !to && isNonZeroHex(calldata)
   const isSend = isNonZeroHex(value)
-  const isContractCall = recipientType !== 'external'
+  const isContractCall = calldata !== '0x'
```

Comment:
I think we have this check twice now, on line 101 and line 109

Reply:
You could include data in a tx to an external account and this would be incorrect right? The `recipientType` checks to see if there's code deployed at that address so maybe we could use both of them together.